test/gurka/test_avoids.cc

```diff
@@ -0,0 +1,287 @@
+#include "gurka.h"
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/multi/geometries/register/multi_polygon.hpp>
+#include <gtest/gtest.h>
+
+#include "baldr/graphconstants.h"
+#include "baldr/graphreader.h"
+#include "loki/polygon_search.h"
+#include "midgard/pointll.h"
+#include "mjolnir/graphtilebuilder.h"
+#include "worker.h"
+
+using namespace valhalla;
+namespace bg = boost::geometry;
+namespace vm = valhalla::midgard;
+namespace vl = valhalla::loki;
+
+namespace {
+rapidjson::Value get_avoid_locs(const std::vector<vm::PointLL>& locs,
+                                rapidjson::MemoryPoolAllocator<>& allocator) {
+  rapidjson::Value locs_j(rapidjson::kArrayType);
+  for (auto& loc : locs) {
+    rapidjson::Value p(rapidjson::kObjectType);
+    p.AddMember("lon", loc.lng(), allocator);
+    p.AddMember("lat", loc.lat(), allocator);
+    locs_j.PushBack(p, allocator);
+  }
+
+  return locs_j;
+}
```

Comment:
the current logic for adding costing options doesn't accommodate for other value types than `string` (ie I need to add `double` as value with the pointer syntax). I didn't wanna mess with that, so I basically roll my own here. ain't saying it's prettier than before, but at least it's only ugly in a very confined space..